test_runner/regress/test_lfc_prewarm.py

```diff
@@ -0,0 +1,40 @@
+import time
+
+from fixtures.log_helper import log
+from fixtures.neon_fixtures import NeonEnv
+
+
+def test_lfc_prewarm(neon_simple_env: NeonEnv):
+    env = neon_simple_env
+    n_records = 1000000
+
+    endpoint = env.endpoints.create_start(
+        branch_name="main",
+        config_lines=[
+            "autovacuum = off",
+            "shared_buffers=1MB",
+            "neon.max_file_cache_size=1GB",
+            "neon.file_cache_size_limit=1GB",
+            "neon.file_cache_prewarm_limit=1000",
+        ],
+    )
+    conn = endpoint.connect()
+    cur = conn.cursor()
+    cur.execute("create extension neon")
+    cur.execute("create table t(pk integer primary key, payload text default repeat('?', 128))")
+    cur.execute(f"insert into t (pk) values (generate_series(1,{n_records}))")
+
+    endpoint.stop()
+    endpoint.start()
+
+    time.sleep(5)  # give prewarm BGW some time to proceed
```

Comment:
I think it would be best if we were able to wait for this to warming to complete, as this 5s might not be enough under a different test suite ordering. This would help with the variable load of test suite on runners.

Perhaps the most natural way would be an sql function we could select over?

Reply:
I do not think that some extra API function is needed for waiting until LFC is prewarmed.
The main idea of prewarm is to perform it in background.
But I agree with you that it will be better to make test more reliable and less fluky.
This is why I replace 5 seconds sleep with the loop which checking number of prewired pages until it reaches specified threshold. Also it should reduce test time.